src/lxc/lsm/apparmor.c

```diff
@@ -439,13 +440,17 @@ static char *apparmor_process_label_get(struct lsm_ops *ops, pid_t pid)
 	if (fd_label < 0)
 		return NULL;
 
-	ret = fd_to_buf(fd_label, &label, &len);
+	ret = fd_to_buf(fd_label, &buf, &len);
 	if (ret < 0)
 		return NULL;
 
 	if (len == 0)
 		return NULL;
 
+	label = malloc(len + 1);
+	memcpy(label, buf, len);
```

Comment:
Hm, we check all allocations for errors. Can you do this here too, please? :)

Reply:
I was so excited that ASan stopped complaining about `strcspn` I didn't notice I forgot to check that. Done. Thanks!